spec/controllers/projects_controller_spec.rb

```diff
@@ -210,4 +182,52 @@
       end
     end
   end
+
+  context "when the project is archived" do
+    before do
+      project.toggle_archived!
+    end
+
+    it "doesn't allow me to edit the project" do
+      get :edit, params: {id: project.id}
+      expect(response).to redirect_to project_path(project)
+    end
+
+    it "doesn't allow me to update the project" do
+      put :update, params: {id: project.id, project: {title: "New Project Title"}}
+      expect(response).to redirect_to project_path(project)
```

Comment:
maybe we can check that `project.title` didn't change here? because the update action, when permitted, also redirects to `project_path(project)`, this test will still pass if somebody breaks the code of the controller